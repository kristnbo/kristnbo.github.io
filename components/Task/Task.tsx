import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';

import { colors } from '../../assets/styles';

interface ITaskProps {
    completed: boolean;
    isBadgeFirst?: boolean;
    img: string;
    imgPlaceholder: string;
    intlPrefix: string;
    className?: string;
}

const Container = styled.section<{ completed: boolean }>`
    display: flex;
    flex-direction: column;
    width: 100%;
    padding: 1rem 20%;
    box-shadow: 0px -1px 0px 0px ${colors.fontColorLight};
    background-color: ${props =>
        props.completed ? colors.greenSuccess : colors.redUnsuccessful};
`;

const BadgeContainer = styled.div<{ reverseOrder: boolean }>`
    display: flex;
    width: 100%;
    flex-direction: ${props => (props.reverseOrder ? 'row-reverse' : 'row')};
    margin-top: 1rem;
`;

const Badge = styled.img<{ reverseOrder: boolean }>`
    height: 10rem;
    margin: ${props => (props.reverseOrder ? '0 0 0 1rem' : '0 1rem 0 0')};
`;

const Div = styled.div`
    height: min-content;
    align-self: center;
`;

const H3 = styled.h3<{ reverseOrder: boolean }>`
    font-size: 2rem;
    color: ${colors.fontColorLight};
    margin-bottom: 1.5rem;
`;

const P = styled.p`
    color: ${colors.fontColorLight};
    margin: auto 0;
`;

const Task: React.FC<ITaskProps> = props => {
    return (
        <Container completed={props.completed} className={props.className}>
            <BadgeContainer reverseOrder={!props.isBadgeFirst}>
                <Badge
                    reverseOrder={!props.isBadgeFirst}
                    src={props.completed ? props.img : props.imgPlaceholder}
                />
                <Div>
                    <H3 reverseOrder={!props.isBadgeFirst}>
                        <FormattedMessage id={`${props.intlPrefix}.title`} />
                    </H3>
                    <P>
                        <FormattedMessage id={`${props.intlPrefix}.text`} />
                    </P>
                </Div>
            </BadgeContainer>
        </Container>
    );
};

export { Task };
