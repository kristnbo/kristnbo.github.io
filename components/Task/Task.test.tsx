import * as React from 'react';

import { render } from '../../../test-utils/testUtil';
import { Task } from './Task';

import testImg from '../../assets/images/deployment.svg';

describe(Task, () => {
    test('snapshot', () => {
        const wrapper = render(
            <Task
                img={testImg}
                imgPlaceholder={testImg}
                intlPrefix="task.test"
                completed={true}
            />
        );
        expect(wrapper.asFragment()).toMatchSnapshot();
    });
});
