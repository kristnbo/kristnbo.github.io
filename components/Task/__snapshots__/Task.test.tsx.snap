// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 1rem 20%;
  box-shadow: 0px -1px 0px 0px #4a4a4a;
  background-color: #85CC7E;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  margin-top: 1rem;
}

.c2 {
  height: 10rem;
  margin: 0 0 0 1rem;
}

.c3 {
  height: -webkit-min-content;
  height: -moz-min-content;
  height: min-content;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c4 {
  font-size: 2rem;
  color: #4a4a4a;
  margin-bottom: 1.5rem;
}

.c5 {
  color: #4a4a4a;
  margin: auto 0;
}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
      />
      <div
        class="c3"
      >
        <h3
          class="c4"
        >
          Better safe than sorry!
        </h3>
        <p
          class="c5"
        >
          Quality assurance is key! Write the neccessary commands to automatically test your code. Each time you push new code to a specific branch, Github actions ought to run all unit tests in the code.
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
